src/components/CodeHighlight/Languages/langJSON.js

```diff
@@ -0,0 +1,19 @@
+export const langJSON = (code, styles) => {
+  const codeString = typeof code === 'string' ? code : JSON.stringify(code, null, 2);
+  const json = codeString.replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;');
+  const re = /("(\\u[a-zA-Z0-9]{4}|\\[^u]|[^\\"])*"(\s*:)?|\b(true|false|null)\b|-?\d+(?:\.\d*)?(?:[eE][+-]?\d+)?)/g;
```

Comment:
Please add a comment for what this regexp is doing as it is hard to reason about it.
Maybe some [tip](https://stackoverflow.com/questions/15463257/commenting-regular-expressions) from this would help.

Reply:
Done